bundle/jsky.app.ot/src/main/scala/jsky/app/ot/gemini/editor/targetComponent/details2/MagnitudeEditor.scala

```diff
@@ -0,0 +1,374 @@
+package jsky.app.ot.gemini.editor.targetComponent.details2
+
+import java.awt.{GridBagConstraints, Insets, GridBagLayout, Component, Dimension}
+import java.awt.event.{ActionEvent, ActionListener}
+import java.beans.{PropertyChangeEvent, PropertyChangeListener}
+import java.text.{NumberFormat, DecimalFormat}
+
+import javax.swing.{SwingUtilities, JScrollPane, BorderFactory, JPanel, DefaultComboBoxModel, JFormattedTextField, JLabel, JList, ListCellRenderer, JButton, JComboBox, JTextField}
```

Comment:
This is a direct port of the Java code, which is pretty complicated. So I didn't want to rewrite the GUI logic.